test/integration/fake_upstream.h

```diff
@@ -291,6 +295,12 @@ class FakeUpstream : Logger::Loggable<Logger::Id::testing>, public Network::Filt
   bool createFilterChain(Network::Connection& connection) override;
   void set_allow_unexpected_disconnects(bool value) { allow_unexpected_disconnects_ = value; }
 
+protected:
+  std::mutex lock_;
```

Comment:
Can you add a comment on what this mutex protects? Do you think there is value in using the TSAN annotations to validate this (given it's only test code)?

Reply:
Yeah, I was overenthusiastic.  Commented but not guarded until we switch to abseil mutex :-)